File: Data Structures & Algorithms/largest-unique-number/submission-1.py
class Solution:
    def largestUniqueNumber(self, nums: List[int]) -> int:
        my_hash = {} # key: number, value = frequency 

        for num in nums: 
           my_hash[num] = my_hash.get(num, 0) + 1
        
        largest = -1
        for num in my_hash:
            if my_hash[num] == 1:
                largest = max(largest, num)

        return largest
        
    

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int[] numbers = new int[nums.length];
        
        for (int i = 0; i < nums.length; i++)
        {
            int product = 1; 
            int j = 0; 
            while (j < nums.length) {
                if (j != i)
                {
                    product *= nums[j];

                }
                j++; 
            }

            numbers[i] = product; 
        }

        return numbers; 

    }
}  
